index/functions.h

```diff
@@ -50,7 +51,19 @@ struct IndexFunction
   int flags;                 ///< Prerequisites for the function, e.g. #CHECK_IN_MAILBOX
 };
 
+/**
+ * enum ResolveMethod - How to advance the cursor
+ */
+enum ResolveMethod
+{
+  RESOLVE_NEXT_EMAIL,        ///< Next email, whatever its state
+  RESOLVE_NEXT_UNDELETED,    ///< Next undeleted email
+  RESOLVE_NEXT_THREAD,       ///< Next top-level thread
+  RESOLVE_NEXT_SUBTHREAD,    ///< Next sibling sub-thread
+};
+
 int index_function_dispatcher(struct MuttWindow *win, int op);
+bool resolve_email(struct Menu *menu, struct IndexSharedData *shared, enum ResolveMethod rm);
```

Comment:
Do you plan on using this from outside index/functions.c?

Reply:
Good point.
I thought I might need it in the Postponed, but I'll probably upgrade that to use the index function dispatcher.

I'll internalise `resolve_email()`.